src/components/RoomView/setupGame.jsx

```diff
@@ -0,0 +1,77 @@
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { setWordsInGame } from "../../store/wordsInGameSlice";
+import { useDispatch, useSelector } from "react-redux";
+import Board from "./Board";
+function SetupGame() {
+  const [wordpacks, setWordpacks] = useState([]);
+  const [selectedWordPackId, setSelectedWordPackId] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const words = useSelector((state) => state.wordsInGame);
+  const dispatch=useDispatch()
+//   //----------------fet all packs for users to select from-----------------//
+  const fetchWordPacks = async () => {
+    setIsLoading(true);
+    const { data } = await axios.get("/api/wordpack");
+    setWordpacks(data);
+
+    setIsLoading(false);
+  };
+  useEffect(() => {
+    fetchWordPacks();
+  }, []);
+
+  //------------------functions to handle selection---------------------//
+
+  const handleWordPackSelection = (event) => {
+    //if event.target.value is already in the array, we delete the already existed one in the array and return
+    if(selectedWordPackId.indexOf(event.target.value)>-1){return selectedWordPackId.splice(selectedWordPackId.indexOf(event.target.value),1)
+    }
+    // if event.target.value is not in the array, we add it in
+    if(selectedWordPackId.indexOf(event.target.value)<0)
```

Comment:
wouldn't it suffice to just use 'else' instead of this if statement?

Reply:
sure, I am not confident with my if else so I used if.